### ISSUE TITLE
Strange "MissingInherits" thrown when using ${BUILD_DIR} of cmake.eclass

### ISSUE THREAD

**Issue by littlewu2508** [pkgcore/pkgcheck#440]
Using pkgcheck 0.10.14, I observed a QA warning on sci-libs/rocBLAS-5.0.2-r2:

```
sci-libs/rocBLAS
  MissingInherits: version 5.0.2-r2: python-r1: missing inherit usage: 'BUILD_DIR', line 96
```
BUILD_DIR is used, and set in cmake.eclass. It does inherit cmake, but still gets MissingInherits warning. Further, it suggests to inherit python-r1 (I guess that's its suggestion), but the ebuild does not use any feature in python-r1.eclass.

**Comment by arthurzam** [comment]
Hi. I checked this because I'm searching for false positives of this check, and hoped you found one, but:

1. I checked `sci-libs/rocBLAS` in `::gentoo` tree, but didn't get any such result.
2. When I checked `cmake.eclass`, I see that it declares `BUILD_DIR` quite well, which result in `pkgcheck` loading it perfectly fine into the check (I checked it in the debugging). So if this failed this is much worse then I anticipated at first.

So, as more info, could you point at which commit it happens to you? Even best if you could point with github commit link to ebuild.

Thank you!

**Comment by thesamesam** [comment]
I'm completely guessing but given that OP is working on a rocm eclass (https://github.com/gentoo/gentoo/pull/26784), there may be some interaction with that.

**Comment by littlewu2508** [comment]
> I'm completely guessing but given that OP is working on a rocm eclass ([gentoo/gentoo#26784](https://github.com/gentoo/gentoo/pull/26784)), there may be some interaction with that.

Probably, but I'm running pkgcheck` on ::gentoo tree as well (at commit 8190c69ccc4082ae04a23f2917d1e3b55e28a6a9)

Turns out its related with cache. If I specify `--cache-dir <another-dir>`, the warning disappears. So I suppose this means we can move focus on  caching system? However this is hard to reproduce, because I didn't notice when the cache is generated.

**Comment by littlewu2508** [comment]
I tried to copy the `eclass.pickle` that leads into this behavior onto another machine. But on another machine, the eclass cache is always  regenerated despite I made the two gentoo repo on each machine syncronized (and with the same pkgcheck version).

[eclass.pickle.gz](https://github.com/pkgcore/pkgcheck/files/9482053/eclass.pickle.gz)

I uploaded this cache here, and If interested, one can dig into it. I will provide more information I can. Currently I suspect the cache is generated when I developing rocm.eclass and becomes malformed. This should be rare, not affecting most ebuid developing.
